scripts/conflicts/diff.js

```diff
@@ -22,35 +28,94 @@ console.log(`Generating diffs for conflicts on ${server}`);
 
 const DB = new PouchDB(server);
 
-DB.query('medic-conflicts/conflicts')
+const writeToFile = function(path, content) {
+  content  = JSON.stringify(content, null, 3); 
+  fs.writeFile(path, content, (err) => {
+    if(err){
+      console.log(err);
+    }
+  });
+};
+
+const createDirectory = function(conflictDirectoryPath) {
+  if (!fs.existsSync(conflictDirectoryPath)){
+    try {
+      fs.mkdirSync(conflictDirectoryPath);
+    } catch(err){
+      console.log(err);
+    }
+  }
+};
+
+const isDiectoryEmpty = function isEmpty(path) {
+  return fs.readdirSync(path).length === 0;
+};
+
+const getConflictDirectoryPath = function (r, mainConflictFilesDirectory) {
+  let conflictDirectoryName = r.id;
+  if (r.id.includes('target')) {
+    conflictDirectoryName = r.id.split(':')[0];
+  } else { // if it is just a contact doc use the doc Id to create folder
+    conflictDirectoryName = r.id;
+  }
+  //The doc id that is used for foldername might have characters that cannot 
+  // be used to create folders. Hence we are replacing illegeal characters with space. 
+  // Reverse this process to search for doc in DB.
+  conflictDirectoryName = conflictDirectoryName.replace(/[/\\?%*:|"<>]/g, ' ');
+  const conflictDirectoryPath = `${mainConflictFilesDirectory}/${conflictDirectoryName}`;
```

Comment:
Please always use `path.resolve` or `path.join` instead of concatenating strings when creating paths. 
```suggestion
  const conflictDirectoryPath = path.join(mainConflictFilesDirectory, conflictDirectoryName);
```